access/client.go

```diff
@@ -0,0 +1,261 @@
+package access
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/safing/portbase/api"
+	"github.com/safing/portbase/database"
+	"github.com/safing/portbase/formats/dsd"
+	"github.com/safing/portbase/log"
+	"github.com/safing/spn/access/account"
+)
+
+// https://account.safing.io/v1/authenticate
+
+const (
+	AccountServer   = "https://account.safing.io"
+	LoginPath       = "/v1/authenticate"
+	UserProfilePath = "/v1/user_profile"
+)
+
+var (
+	accountClient     = &http.Client{}
+	clientRequestLock sync.Mutex
+)
+
+func login(ctx context.Context, username, password string) (user *UserRecord, code int, err error) {
+	clientRequestLock.Lock()
+	defer clientRequestLock.Unlock()
+
+	// Get previous user.
+	previousUser, err := GetUser()
+	if err != nil {
+		previousUser = nil
+	}
```

Comment:
I guess we should handle the error here only in case it's a "not-logged-in" error and not anything else. At least, we should log the error then